### PR TITLE
buildkite-agent-metrics: init at 5.2.1, nixos/prometheus-buildkite-agent-exporter: init

### PR DESCRIPTION
**Pull request by cole-h**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by grahamc — COMMENTED** [review]
Looks good from here, but let's deploy and verify it internally.

**Review by SuperSandro2000 — APPROVED** [review]

File: nixos/modules/services/monitoring/prometheus/exporters/buildkite-agent.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    serviceConfig = {
+      DynamicUser = false;
+      RuntimeDirectory = "buildkite-agent-metrics";
+      ExecStart =
+        let
+          queues = concatStringsSep " " (map (q: "-queue ${q}") cfg.queues);
+        in
+        ''
+          ${pkgs.buildkite-agent-metrics}/bin/buildkite-agent-metrics \
+            -token "$(cat ${toString cfg.tokenPath})" \
```
Comment (grahamc) [inline nixos/modules/services/monitoring/prometheus/exporters/buildkite-agent.nix:55]: It is actually dangerous to pass the token over argv, so let's set BUILDKITE_AGENT_TOKEN instead in the environment.

**Comment by cole-h** [comment]
Converted to draft because `BUILDKITE_AGENT_TOKEN` doesn't work for the metrics tool, only the lambda tool. We don't want to leak the file in argv, so that's the only way forward. Issue filed upstream: https://github.com/buildkite/buildkite-agent-metrics/issues/115.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130291` at 92fd0056 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>buildkite-agent-metrics</li>
    <li>nixos-install-tools</li>
  </ul>
</details>

**Comment by cole-h** [comment]
Upstream was incredibly speedy in merging my patch fixing the `BUILDKITE_AGENT_TOKEN` issue -- I've added a patch in the latest few fixups in order to incorporate that.

**Review by grahamc — APPROVED** [review]

**Comment by cole-h** [comment]
@grahamc deployed this to one of his machines and it worked great:
```
buildkite_queues_scheduled_jobs_count{queue="default"} 0
buildkite_queues_waiting_jobs_count{queue="default"} 0
buildkite_total_busy_agent_count 1
```

**Review by grahamc — APPROVED** [review]

**Review by grahamc — APPROVED** [review]

**Pull request merged**